src/components/App.jsx

```diff
@@ -89,42 +90,74 @@ const theme = createMuiTheme({
   }
 });
 
-const PrivateRoutes = () => (
+function PrivateRoute ({component: Component, user, path, ...rest}) {
+  return (
+    <Route
+      path={path}
+      {...rest}
+      render={(props) => user
+        ? <Component {...props} />
+        : <Redirect to={{pathname: '/', state: {from: props.location}}} />}
+    />
+  )
+}
+
+const Routes = ({state}) => (
   <Switch>
-    <Redirect exact from="/" to="/events" />
-    <Route exact path="/events" component={EventsSection} />
-    <Route exact path="/settings" component={SettingsSection} />
-    <Route path="/events/:eventId/performances" component={PerformancesSection} />
-    <Route path="/events/:eventId/performance/:performanceId/adjudications" component={AdjudicationsSection} />
+    <Route exact path="/" component={SignInPage}  />
+    <PrivateRoute user={state.user} path="/events/:eventId/performances" component={PerformancesSection} />
+    <PrivateRoute user={state.user} path="/events/:eventId/performance/:performanceId/adjudications" component={AdjudicationsSection} />
+    <PrivateRoute user={state.user} exact path='/events' component={EventsSection} />
+    <PrivateRoute user={state.user} exact path="/settings" component={SettingsSection} />
     <Route component={Landing} />
   </Switch>
 );
 
 const PublicRoutes = () => (
   <Switch>
-    <Route exact path="/" component={Landing} />
+    <Route exact path="/" component={SignInPage} />
     <Route exact path={routes.SIGN_IN} component={SignInPage} />
     <Route component={Landing} />
   </Switch>
 );
 
 export default class App extends React.Component {
-  state = {};
+  constructor(props) {
+    super(props);
+
+    this.state={
+      user: auth.currentUser,
+      loading: true
+    }
+  }
+
+  componentDidMount() {
+    auth.onAuthStateChanged((user) => {
+      this.setState({ user: user, loading: false });
+    });
+  }
+
 
   render() {
-    // TODO: check user login from Redux
-    const authenticated = true;
+    const { user, loading } = this.state;
     return (
       <MuiThemeProvider theme={theme}>
         <Router>
-          {authenticated ? (
             <div>
-              <Header />
-              <PrivateRoutes />
+              { !loading && false && <Header /> }
```

Comment:
This needs to be modified later to contains `!loading && user &&...` Maintained as this for testing the UI.